paddle/fluid/operators/matmul_op.cc

```diff
@@ -54,13 +228,69 @@ class MatMulKernel : public framework::OpKernel<T> {
     auto *out = context.Output<framework::Tensor>("Out");
     out->mutable_data<T>(context.GetPlace());
 
+    auto transpose_x = context.Attr<bool>("transpose_X");
+    auto transpose_y = context.Attr<bool>("transpose_Y");
+    auto dim_x = x.dims();
+    auto dim_y = y.dims();
+
+    int ld_x = 0, ld_y = 0, ld_out = 0;
+    std::vector<const T *> a_array, b_array;
```

Comment:
maybe could use const T (*p)[n] to avoid using std:vector? so can reuse matmul in blas?

Reply:
Now Bingyang are ready to re-implement the pass in future. This PR will be aborted.